test/Microsoft.AspNetCore.Razor.Test/CodeGenerators/DecorateChunksTestCodeGenerator.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Chunks;
+using Microsoft.AspNetCore.Razor.CodeGenerators;
+
+namespace Microsoft.AspNetCore.Razor.Test.CodeGenerators
+{
+    public class DecorateChunksTestCodeGenerator : CSharpCodeGenerator 
+    {
+        public enum GeneratorType
+        {
+            Add = 0,        // Add duplicates the first chunk in the list.
+            Clear = 1,      // Clear removes all chunks.
+            Default = 2     // Default does not modify chunks.
+        }
+
+        private readonly int _generatorType;
+
+        public DecorateChunksTestCodeGenerator(CodeGeneratorContext context, int type) : base(context)
```

Comment:
Why isn't `type` of type `GeneratorType`?

That said, there isn't so much boilerplate here that 3 separate classes would be unworkable. That should hopefully align with `BuildAfterExecuteContentTestCodeGenerator` and allow you to add just one `[Theory]` and data set to `CSharpCodeGeneratorTest`. (From a test perspective, they're all just `CSharpCodeGenerator` that need to be run and have their output checked.)